framework/doc/content/source/csg/CSGBase.md

```diff
@@ -0,0 +1,663 @@
+
+- `getCSGBase`: get the `CSGBase` object given a parameter name represented as a `std::string` that stores the mesh generator name
+- `getCSGBases`: get the `CSGBase` objects given a parameter name represented as a `std::string` that stores a list of mesh generator names
+- `getCSGBaseByName`: get the `CSGBase` object given a `MeshGeneratorName`
+- `getCSGBasesByName`: get all `CSGBase` objects given a list of `MeshGeneratorName`s
+
+For example:
+
+```cpp
+// get the CSGBase from a different mesh generator and use in this mesh generator
+// other_mg_name is a MeshGeneratorName object
+auto csg_base = getCSGBaseByName(other_mg_name);
+std::unique_ptr<CSG::CSGBase> csg_obj = std::move(*csg_base);
+// csg_obj is now the object that will continue to get updated throughout the generateCSG method.
+```
+
+!alert! note title=Accessing other MeshGenerator objects by name
+
+A [source/meshgenerators/MeshGenerator.md] object(s) can be passed to another mesh generator as input by providing `InputParameters` of type `MeshGeneratorName`.
+See the `ExampleAxialSurfaceMeshGenerator` implementation [below](#example-implementation) for an example of this.
+
+!alert-end!
+
+### Joining Bases
+
+When two or more existing `CSGBase` objects need to be combined to continue to use and update, the `joinOtherBase` method should be used.
+This method is called from another `CSGBase` and at a minimum takes a different existing `CSGBase` object as input.
+There are 3 different behaviors for joining bases that are supported depending on the additional arguments that are passed:
+
+1. No additional arguments: All cells that are in the root universe of the incoming `CSGBase` object will be added to the existing root universe of the current base object, and the root universe from the incoming base will no longer exist.
+
+```cpp
+// get a list of bases associated with a list of mesh generator names
+const auto csg_bases = getCSGBases("input_mg");
+// first mesh generator base will be used as the main and all others will be joined to this one.
+std::unique_ptr<CSG::CSGBase> csg_obj = std::move(*csg_bases[0]);
+for (unsigned int i = 1; i < csg_bases.size(); ++i)
+{
+  // get the specific base object from the input and join to the first
+  std::unique_ptr<CSG::CSGBase> inp_csg_obj = std::move(*csg_bases[i]);
+  // all cells in root for inp_csg_obj are moved to the root of the current csg_obj
+  csg_obj->joinOtherBase(inp_csg_obj);
+}
+```
+
+2. One new root universe name (`new_root_name_join`): All cells in the root universe of the incoming base will be used to create a new universe of the name specified by the `new_root_name_join` parameter. These cells will *not* be added to the existing root universe. This new universe will be added as a new non-root universe in the existing base object. *This newly created universe will not be connected to the root universe of the existing `CSGBase` object by default.*
```

Comment:
this seems equivalent to taking the root universe of the second CSGBase and placing it as an universe in the first one?

Reply:
Are you asking why this method needs to exist when you could just do what you suggest?

On the surface, yes this method does what you are describing. However, the `joinOtherBase` methods are required to be able to fully track all the data down to the CSGCell and CSGSurface level of information. If you simply add the universe from one base to another, you will have the list of cell names associated with it, but you will not have the CSGCell objects or the CSGSurface objects associated with the original base. These join methods not only join/track universes, but they also make sure all the cells and surfaces are also transferred properly too. The reason for having the three different join methods is because you could handle the universes in different ways.